src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -940,43 +950,66 @@ - (void)_handleSubscriptionError:(NSError *)error
 
 // This method is used for signaling whether to use the subscription pool. This functions as
 // a heuristic for whether to throttle subscriptions to the device via a pool of subscriptions.
-// If products appear that have both Thread and Wifi enabled but is primarily on wifi, this
+// If products appear that have both Thread and Wifi enabled but are primarily on wifi, this
 // method will need to be updated to reflect that.
 - (BOOL)_deviceUsesThread
 {
     os_unfair_lock_assert_owner(&self->_lock);
 
-    // Device is thread-enabled if there is a Thread Network Diagnostics cluster on endpoint 0
-    for (MTRClusterPath * path in [self _knownClusters]) {
-        if (path.endpoint.unsignedShortValue != 0) {
-            continue;
+#ifdef DEBUG
+    id testDelegate = _weakDelegate.strongObject;
+    if (testDelegate) {
+        if ([testDelegate respondsToSelector:@selector(unitTestPretendThreadEnabled:)]) {
```

Comment:
Might be worth a comment explaining why it's OK to not do this on the delegate queue.  Especially because other unit test callbacks we _do_ go out of our way to call on that queue.